facade/src/main/java/org/jboss/pnc/facade/impl/DeliverablesAnalyzerInvokerImpl.java

```diff
@@ -0,0 +1,107 @@
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.MediaType;
+import java.net.URI;
+import java.util.HashSet;
+import java.util.Set;
+
+@ApplicationScoped
+public class DeliverablesAnalyzerInvokerImpl implements DeliverablesAnalyzerInvoker {
+
+    private static final Logger log = LoggerFactory.getLogger(DeliverablesAnalyzerInvokerImpl.class);
+
+    private UserService userService;
+
+    private BpmManager bpmManager;
+
+    private GlobalModuleGroup globalConfig;
+
+    private BpmModuleConfig bpmConfig;
+
+    // TODO enter the actual endpoint once ready
+    private String callbackUrlTemplate = "%s/product-milestones/%s/TODO";
+
+    @Inject
+    public DeliverablesAnalyzerInvokerImpl(
+            UserService userService,
+            BpmManager bpmManager,
+            GlobalModuleGroup globalConfig,
+            BpmModuleConfig bpmConfig) {
+        this.userService = userService;
+        this.bpmManager = bpmManager;
+        this.globalConfig = globalConfig;
+        this.bpmConfig = bpmConfig;
+    }
+
+    @Override
+    public void startAnalysis(String milestoneId, DeliverablesAnalysisRequest request) {
+        String accessToken = userService.currentUserToken();
+
+        String actualEndpoint = String.format(callbackUrlTemplate, globalConfig.getPncUrl(), milestoneId);
+        URI callbackURI = URI.create(actualEndpoint);
+
+        Set<Request.Header> headers = new HashSet<>();
+        addCommonHeaders(headers, accessToken);
+        addMDCHeaders(headers);
+
+        Request callback = new Request(Request.Method.POST, callbackURI, headers);
+
+        try (RestConnector restConnector = new RestConnector(bpmConfig)) {
+            AnalyzeDeliverablesBpmRequest bpmRequest = new AnalyzeDeliverablesBpmRequest(
+                    request.getSourcesLink(),
+                    null);
+            AnalyzeDeliverablesTask analyzeTask = new AnalyzeDeliverablesTask(bpmRequest, accessToken, callback);
+            analyzeTask.setTaskId(bpmManager.getNextTaskId());
+            analyzeTask.setGlobalConfig(globalConfig);
+            analyzeTask.setJsonEncodedProcessParameters(false);
+            restConnector.startProcess(
+                    bpmConfig.getAnalyzeDeliverablesBpmProcessId(),
+                    analyzeTask.getExtendedProcessParameters(),
+                    accessToken);
+        } catch (CoreException | ProcessManagerException e) {
+            log.error("Error trying to start analysis of deliverables task for milestone: {}", milestoneId, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void addMDCHeaders(Set<Request.Header> headers) {
+        headersFromMdc(headers, MDCHeaderKeys.REQUEST_CONTEXT);
+        headersFromMdc(headers, MDCHeaderKeys.PROCESS_CONTEXT);
+        headersFromMdc(headers, MDCHeaderKeys.TMP);
+        headersFromMdc(headers, MDCHeaderKeys.EXP);
```

Comment:
I guess tmp and exp have no meaning in this case.
Where is the PROCESS_CONTEXT set ?

Reply:
These headers are for the callback that BPM will invoke once DelAnal is done. The other MDC parameters which we discussed to duplicate, have in its own class and are directly for BPM, are set here https://github.com/project-ncl/pnc/pull/3461/files#diff-d9880356c1a2790e5f7caab27c18a69df6cb556a67c2f00631aa9aa4c458b193R83

Reply:
It is set now in PMEndpoint